extension/src/popup/components/accountHistory/HistoryItem/index.tsx

```diff
@@ -178,64 +181,85 @@ export const HistoryItem = ({
   /* eslint-disable react-hooks/exhaustive-deps */
   const translations = useCallback(t, []);
 
+  useEffect(() => {
+    const map = async () => {
+      const historyItemData = await mapHistoryItemData({
+        operation,
+        publicKey,
+        networkDetails,
+      });
+
+      console.log("historyItemData", historyItemData);
+    };
+
+    map();
+  }, []);
+
   useEffect(() => {
     const buildHistoryItem = async () => {
       setIsLoading(true);
 
-      if (type === Horizon.HorizonApi.OperationResponseType.createAccount) {
+      const historyItemData = await mapHistoryItemData({
+        operation,
+        publicKey,
+        networkDetails,
+      });
+
+      const IconComponent =
+        Icon[historyItemData.iconString as keyof typeof Icon];
+
+      const ActionIconComponent =
+        Icon[historyItemData.actionIconString as keyof typeof Icon];
+
+      if (isCreateAccountOperation(type)) {
         // If you're not creating an external account then this means you're
         // receiving some XLM to create(fund) your own account
-        const _isRecipient = !isCreateExternalAccount;
-        const paymentDifference = _isRecipient ? "+" : "-";
-        const formattedAmount = `${paymentDifference}${formatAmount(
-          new BigNumber(startingBalance).toString(),
-        )} ${destAssetCode}`;
```

Comment:
we can get rid of all this formatting logic and just use what comes back from the shared lib